src/fu-util.c

```diff
@@ -2136,18 +2137,31 @@ fu_util_download_metadata(FuUtilPrivate *priv, GError **error)
 		FwupdDevice *dev = g_ptr_array_index(devs, i);
 		if (fwupd_device_has_flag(dev, FWUPD_DEVICE_FLAG_SUPPORTED))
 			devices_supported_cnt++;
+		if (fwupd_device_has_flag(dev, FWUPD_DEVICE_FLAG_UPDATABLE))
+			devices_updatable_cnt++;
 	}
 
 	/* TRANSLATORS: success message -- where 'metadata' is information
 	 * about available firmware on the remote server */
 	g_string_append(str, _("Successfully downloaded new metadata: "));
 
-	g_string_append_printf(str,
-			       /* TRANSLATORS: how many local devices can expect updates now */
-			       ngettext("%u local device supported",
-					"%u local devices supported",
-					devices_supported_cnt),
-			       devices_supported_cnt);
+	if (devices_supported_cnt > 0) {
+		g_string_append_printf(
+		    str,
+		    /* TRANSLATORS: how many local devices can expect updates now */
+		    ngettext("%u local device supported, with published updates",
+			     "%u local devices supported, with published updates",
+			     devices_supported_cnt),
+		    devices_supported_cnt);
+	} else {
+		g_string_append_printf(
+		    str,
+		    /* TRANSLATORS: message to tell user not to expect updates from remote */
+		    ngettext("%u local device supported, with no published updates",
+			     "%u local devices supported, with no published updates",
+			     devices_updatable_cnt),
+		    devices_updatable_cnt);
+	}
```

Comment:
do you think we are trying to encode two different things into one string? At the end of the day this is for the refresh action, so we're telling the user how many things *matched*. Perhaps we could export it like `%u of %u local devices are supported` -- so in the case where you have no supported devices you'd see `0 of 7 local devices are supported` and for the common case you'd see something like `5 of 7 local devices are supported`.